scripts/sudo.ps1

```diff
@@ -3,85 +3,86 @@
 
 # open question - Should -NoProfile be used when invoking PowerShell
 BEGIN {
-	if ($__SUDO_TEST -ne $true) {
-		$SUDOEXE = "sudo.exe"
-	}
-	else {
-		if ($null -eq $SUDOEXE) {
-			throw "variable SUDOEXE has not been set for testing"
-		}
-	}
+    if ($__SUDO_TEST -ne $true) {
+        [System.String]$SUDOEXE = 'sudo.exe'
+    }
+    else {
+        if ($null -eq $SUDOEXE) {
+            throw [System.InvalidOperationException] 'Variable SUDOEXE has not been set for testing.'
+        }
+    }
 
-	if ([Environment]::OSVersion.Platform -ne "Win32NT") {
-		throw "This script works only on Microsoft Windows"
-	}
+    if ([Environment]::OSVersion.Platform -ne 'Win32NT') {
+        throw [System.PlatformNotSupportedException] 'This script works only on Microsoft Windows'
+    }
 
-	if ($null -eq (Get-Command -Type Application -Name "$SUDOEXE" -ErrorAction Ignore)) {
-		throw "'$SUDOEXE' cannot be found."
-	}
+    if ($null -eq (Get-Command -Type Application -Name "$SUDOEXE" -ErrorAction Ignore)) {
+        throw [System.IO.FileNotFoundException] "'$SUDOEXE' cannot be found."
+    }
 
-	$psProcess = Get-Process -id $PID
-	if (($null -eq $psProcess) -or ($psProcess.Count -ne 1)) {
-		throw "Cannot retrieve process for '$PID'"
-	}
+    $psProcess = Get-Process -Id $PID
+    if (($null -eq $psProcess) -or ($psProcess.Count -ne 1)) {
+        throw [System.InvalidOperationException] "Cannot retrieve process for '$PID'"
+    }
 
-	$thisPowerShell = $psProcess.MainModule.FileName
-	if ($null -eq $thisPowerShell) {
-		throw "Cannot determine path to '$psProcess'"
-	}
+    $thisPowerShell = $psProcess.MainModule.FileName
+    if ($null -eq $thisPowerShell) {
+        throw [System.InvalidOperationException] "Cannot determine path to '$psProcess'"
+    }
 
-	function convertToBase64EncodedString([string]$cmdLine) {
-		$bytes = [System.Text.Encoding]::Unicode.GetBytes($cmdLine)
-		[Convert]::ToBase64String($bytes)
-	}
+    function ConvertTo-Base64EncodedString([System.String]$cmdLine) {
```

Comment:
Why not use `[string]`, the builtin type?

Reply:
Because it eliminates confusion when we explicitly type out the fully qualified type name instead of using type accelerator/alias.

Reply:
I understand the other argument (in fact it's a fantastic one!), but this one seems to be just an opinion. (It's easily countered with "How does it eliminate confusion?", etc., for instance.) I'm not really meaning to press this further, however.

Reply:
The confusion elimination is by being explicit instead of implicit

like in this example

```powershell
$CSharpCode = @'
namespace MyNamespace
{
    public class String
    {
        public string Value { get; set; }
    }
}
'@
Add-Type -TypeDefinition $CSharpCode
Set-Alias -Name String -Value MyNamespace.String
$myStringObject = [String]::new()
$myStringObject.Value = 'Hello'
$myStringObject.Value
```

That doesn't work unless i disambiguate it by using this

```powershell
$myStringObject = [MyNamespace.String]::new()
```

since both System and my custom namespace have the same String class.